### PR TITLE
Write some Summary text

### PR DESCRIPTION
**Pull request by chutten**
This adds some summary text for categorical and boolean data.

Let the bikeshed begin.

### CODE REVIEW

**Review by ecomerford — COMMENTED** [review]
This looks pretty good to me. I actually like the idea of having the categorical copy show the most/least common, but I guess we should also let Georg weigh in. I don't have any notes for the code itself, it looks pretty solid.

**Comment by chutten** [comment]
This should also fix #36

**Review by georgf — CHANGES_REQUESTED** [review]
I'll focus on the design review here for now.

On a meta note, i find it hard to review either design or code when they are mixed together.
For future work lets go for separate design conversations please. Github isn't necessarily the best place to iterate on phrasing or concepts that we haven't discussed before, a Google doc could be much faster here.
For example it's hard to propose completely different approaches here.

**Both booleans and categorical**

I'm missing a notion of "change since last version" for both categorical and booleans. Is that planned for a follow-up?

The user numbers shown are pretty large and could use "," separator formatting.

**Booleans**

To allow for easy scanning of the headings, let's make these *"Always true"*, *"Ever true"*, *"Never true"*.

For the longer explanations, i would say:
a) We should use one terminology.
b) "Respondents" is an intuitively unfamiliar term to me.
c) The sentence front-loads a lot of numbers and is hard to parse for me.

How about *"X% of users always respond true for {metric}. That is N of M users."*
Etc. for the others.

**Categoricals**

Interesting approach, that makes them definitely more useful than the current display. 
Do we know yet what people look for mostly when evaluating categoricals?
Should we add a task in user tests that cover them?

For the headings, how about using a format like *"CategoryX with Y%"*?
*"PrefBlockedSTS (0.00001393%)"* - do we care about that much precision in the heading? Can we round two 3 decimals or something? Or say "less than X%"?

The lists look like mostly numbers to me at a glance, they could use a bit of wording for understanding them.
- How about using bulleted lists? Is it necessary to use numbered lists here?
- Can we add some wording? E.g. *"76.21% of users reported AlreadyHTTPS. That is N of M users."*
- Ditto about rounding, do we really need that accuracy?

**Comment by chutten** [comment]
> On a meta note, i find it hard to review either design or code when they are mixed together.

Fair. To balance the other side it is sometimes difficult to see what is possible or easy without first prototyping a rough outline. Maybe analogous to the ideation and proposal phases?

At any rate, from a broad perspective I intend to address the smaller things to get this prototype in and then we can have a proper design review of alternatives and discussion about what this means for user study.

>I'm missing a notion of "change since last version" for both categorical and booleans. Is that planned for a follow-up?

This is awaiting #65. I have now filed #69 to formalize the follow-up.

>The user numbers shown are pretty large and could use "," separator formatting.

I agree. (Something I noticed: I'm not sold on the font giving "3" and "5" descenders instead of sitting them on the baseline) I contemplated giving these figures some fixed number of significant digits and then the SI suffix, but I think we want precision in these measurements.

Separators will do the trick, and I'm pretty sure `d3.format` supports them.

> To allow for easy scanning of the headings, let's make these "Always true", "Ever true", "Never true".

Righto, will do.

> How about "X% of users always respond true for {metric}. That is N of M users."

I like it. We're having a conversation on #68 about respondents but by providing `M` we can avoid it entirely here. If we then have a "total clients in this set of dimensions" figure we can add that later without too much trouble. 

>  Do we know yet what people look for mostly when evaluating categoricals?

We do not. At least not formally. From my personal experience with e.g. [TELEMETRY_SEND_FAILURE_TYPE](https://mzl.la/2NsHHnV) the most- and least-common categories were of interest. The shape of the data (order of the categories) is irrelevant, I pretty much just want a table of category and popularity, sorted by popularity (descending).

>  Should we add a task in user tests that cover them?

I leave that to the person writing the user study : D

But in all seriousness, I'm divided on this. On one hand I want to know. I want this to be useful for all data types and for our users, and the only reasonable way to get there is to measure it. However, we have more important things to evaluate (like our entire design methodology) so I'm not sure if we should spend any of this study's time on something this new.

Let's discuss this when we chat about the study.

> For the headings, how about using a format like "CategoryX with Y%"?

Oh, as in "Most Common: CategoryX with Y%"? Yeah, sounds good.

>  do we care about that much precision in the heading?

Likely not, no. 4 [sig figs](https://en.wikipedia.org/wiki/Significant_figures) is what we were using in the distribution view, so I copied it here. We can drop it to three, or...

> Or say "less than X%"?

We can have a sliding scale of the string "less than 0.001%" for < 0.001%, 2 sig figs for < 10, 3 for the rest. (I do a similar thing for the distribution view for low absolute numbers so we don't have "17.00 users" in the hover text)

I'm on the fence if this level of format consistency suggests a utility component for storing the d3 formats we use and applying them or whether d3.format is concise enough to use inline each time. Something to think about later.

> How about using bulleted lists? Is it necessary to use numbered lists here?

We're listing the three most common or three least common, it seemed only right that they should be numbered to emphasize that 1 is the most common, 2 is the second-most common, and so forth. Maybe the top 5 would make that better?

> Can we add some wording? E.g. "76.21% of users reported AlreadyHTTPS. That is N of M users."

It'd be a nice echo of the boolean changes, sure.

> Ditto about rounding, do we really need that accuracy?

For non-headers I feel we want some precision... but if we're including the raw figures (N and M) then we're providing infinite precision in a "build it yourself" kinda way, so maybe we can use the same sliding scale from the headers here. I'll play with it.

---

Thanks for the comprehensive review! I'll see if I can move this prototype towards a more landable direction.

**Comment by chutten** [comment]
...Now that I try implementing this, saying "76.21% of users reported AlreadyHTTPS. That is N of M users." might not be possible or easy or desired depending on how aggregation goes. It may be a "%ge of users' experience" type of aggregation, though I don't think we need to do that for categorical-type data. (I've brought this up in the dataset doc so we can chat with frank about it)

From a comprehesion/intuition point of view, with such an aggregation (% of clients reporting X) we will have the categories almost certainly summing to something more than 100%. It may be confusing to do it this way (or it might not, who can say).

**Comment by chutten** [comment]
RFAL

**Comment by chutten** [comment]
We had an extensive design review ([document](https://docs.google.com/document/d/1PrSPdoVS3e0-vjkKzq1gye00iRz0vfUiiQK-5Aw5rOM/edit#)) which will result in a different Pull Request for the code review of these design changes and the others discussed.

Closing.
